x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/esql/utils/generate_alert_id.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import objectHash from 'object-hash';
+import type { Moment } from 'moment';
+import type * as estypes from '@elastic/elasticsearch/lib/api/types';
+
+import type { CompleteRule, EsqlRuleParams } from '../../../rule_schema';
+import type { SignalSource } from '../../types';
+import type { SuppressionTerm } from '../../utils/suppression_utils';
+/**
+ * generates id for ES|QL alert
+ */
+export const generateAlertId = ({
```

Comment:
Can we add comment here and describe why we need to use hash, and not just provide unique id?

Reply:
https://github.com/elastic/kibana/pull/180927/commits/0cef9076b2b7f2063886955e9ddd823b6ce8eea6